agent/stats/queue.go

```diff
@@ -343,6 +346,11 @@ func (queue *Queue) getCWStatsSet(f getUsageFloatFunc) (*ecstcs.CWStatsSet, erro
 		sum += thisStat
 	}
 
+	// don't emit metrics when sampleCount == 0
```

Comment:
We have the check before this to make sure the queue buffer is not less than 2, do we still need this check for sampleCount here?

Reply:
yes because we skip the metric if it's NaN (see the continue above), so if all the metrics are NaN then we can get down here without having set any values on the cloudwatch metric, and sampleCount will be 0.